name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/wizards/datatransfer/PDFImportAssistant.java

```diff
@@ -0,0 +1,156 @@
+            case "ingdiba": //$NON-NLS-1$
+                return new INGDiBaExtractor(client);
+            case "onvista": //$NON-NLS-1$
+                return new OnvistaPDFExtractor(client);
+            case "sbroker": //$NON-NLS-1$
+                return new SBrokerPDFExtractor(client);
+            case "unicredit": //$NON-NLS-1$
+                return new UnicreditPDFExtractor(client);
+            default:
+                return new AssistantPDFExtractor(client);
+        }
+    }
+
+    public static String DetectBankIdentifier(File PDFfilename) throws IOException  
+    {  
+
+        // Loading an existing document   
+        PDDocument document = PDDocument.load(PDFfilename);  
+
+        // Getting the PDDocumentInformation object  
+        PDDocumentInformation pdd = document.getDocumentInformation();  
+
+        // Getting the PDFTextStripper object  
+        PDFTextStripper textStripper = new PDFTextStripper();  
+        textStripper.setSortByPosition(true);  
+        String text = textStripper.getText(document);  
+        document.close();  
+
+        // PDF import assistant - Level 1 - Allocate by PDF Author  
+        String PDFauthor;  
+        if (pdd.getAuthor() == null) PDFauthor = "";  
+        else PDFauthor = pdd.getAuthor();  
+
+        switch (PDFauthor)  
+        {  
+
+            case "Scalable Capital Vermögensverwaltung GmbH": //$NON-NLS-1$  
+                return "baaderbank";  
+            case "Consorsbank": //$NON-NLS-1$  
+                return "consorsbank";  
+            case "Computershare Communication Services GmbH": //$NON-NLS-1$  
+                return "dab";  
+            case "DKB AG": //$NON-NLS-1$  
+                return "dkb";  
+            case "ING-DiBa": //$NON-NLS-1$  
+                return "ingdiba";  
+
+            default:  
+
+                // PDF import assistant - Precheck if specific securities could cause faulty detection  
+                // ISIN DE0005088108 = Baader Bank Aktie detect the bank identifier "Baader Bank"   
+                Matcher matcherISIN = Pattern.compile("DE0005088108|DE0005428007|DE000CBK1001|FR0000131104|INE007B01023|DE000FTG1111|CH0001351862|CH0001350328|CH0001354296|CH0001352720|CH0001350112|CH0318681860|CH0001343885|FR0000131104|INE007B01023").matcher( text );  
+                int matcherISIN_count = 0;  
+                while (matcherISIN.find())  
+                    matcherISIN_count++;  
+                if (matcherISIN_count != 0) return ("others");  
+
+                // PDF import assistant - Search for the bank identifier  
+                Matcher matcherBankIdentifier = Pattern.compile("Baader Bank|Bank SLM AG|biw AG|BNP Paribas|C O M M E R Z B A N K|comdirect bank|Consorsbank|Cortal Consors|DAB Bank|DEGIRO B.V|Deutsche Bank Privat- und Geschäftskunden AG|FinTech Group Bank AG|ING-DiBa AG|S Broker AG & Co. KG|Scalable Capital|Spar + Leihkasse|UniCredit Bank AG").matcher( text );  
```

Comment:
Das sind doch die "bankIdentifier" die in jedem Extractor definiert werden, oder? Statt die an dieser Stelle noch mal aufzulisten, sollten die besser bei jedem Extractor abgefragt werden. Dann hat man *eine* Datei in der die Bank-spezifischen Änderungen gemacht werden.

Reply:
Richtig,  daß sind die "bankIdentifier" aus den einzelnen Extractor Klassen,  aber ich kann diese wg. fehlenden Java Wissen nicht zur Laufzeit abfragen.